src/UI/Components/TreeTable/TreeRow/CellWithCopyExpert.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useState, MouseEvent, useContext, useEffect } from "react";
+import { Button, Popover, Icon, Modal, Text } from "@patternfly/react-core";
+import { TimesIcon, PencilAltIcon } from "@patternfly/react-icons";
+import { Td } from "@patternfly/react-table";
+import { Environment, Maybe, ParsedNumber } from "@/Core";
+import { AttributeSet } from "@/Core/Domain/ServiceInstanceParams";
+import { DependencyContext } from "@/UI/Dependency";
+import { words } from "@/UI/words";
+import { ConfirmUserActionForm } from "../../../Components/ConfirmUserActionForm";
+import { ToastAlert } from "../../../Components/ToastAlert";
+import { CustomEvent } from "../../ExpertBanner";
+import { TreeTableCellContext } from "../RowReferenceContext";
+import {
+  formatValue,
+  MultiLinkCell,
+  shouldRenderLink,
+  StyledButton,
+  StyledPopoverBody,
+} from "./CellWithCopy";
+import InlineInput from "./InlineInput";
+
+interface Props {
+  className: string;
+  label: string;
+  value: string;
+  hasOnClick?: boolean;
+  serviceName?: string;
+  path: string;
+  instanceId: string;
+  version: ParsedNumber;
+  serviceEntity: string;
+  attributeType: string;
+}
+
+export const CellWithCopyExpert: React.FC<Props> = ({
+  label,
+  value,
+  className,
+  hasOnClick,
+  serviceName,
+  path,
+  instanceId,
+  version,
+  serviceEntity,
+  attributeType,
+}) => {
+  const { commandResolver, environmentHandler } = useContext(DependencyContext);
+  const environment = environmentHandler.useSelected() as
+    | Environment
+    | undefined;
+  const [isExpertMode, setIsExpertMode] = useState(
+    environment?.settings.enable_lsm_expert_mode ? true : false
+  );
+  const [wrapWithPopover, setWrapWithPopover] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [newAttribute, setNewAttribute] = useState<string | boolean | number>(
+    value
+  );
+  const [isInputOpen, setIsInputOpen] = useState(false);
+  const [stateErrorMessage, setStateErrorMessage] = useState<string>("");
+  const { onClick } = useContext(TreeTableCellContext);
+  const trigger = commandResolver.useGetTrigger<"UpdateInstanceAttribute">({
+    kind: "UpdateInstanceAttribute",
+    service_entity: serviceEntity,
+    id: instanceId,
+    version,
+  });
+
+  const onMouseEnter = (event: MouseEvent<HTMLTableCellElement>) => {
+    // Check if overflown
+    if (isInputOpen) return;
+    if (event.currentTarget.offsetWidth < event.currentTarget.scrollWidth) {
+      setWrapWithPopover(true);
+    } else {
+      setWrapWithPopover(false);
+    }
+  };
+  const onSubmit = async () => {
+    let newValue: string | string[] | number | boolean;
+    if (attributeType.includes("string[]")) {
+      newValue = (newAttribute as string).replace(/\s/g, "").split(",");
```

Comment:
isn't newAttribute at that point already an array of strings ?

Reply:
nope input is text, so it gets as something like "123, 123"

Reply:
Ping me when you made the change, the rest is fine, so this is the last open comment